### PR TITLE
60 | Add filters in overview sidebar

### PR DESCRIPTION
**Pull request by kino90**

### CODE REVIEW

File: src/components/layout/SideBar/Overview/OverviewFilters/AssigneeFilter/AssigneeFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      event.stopPropagation();
+
+      if (onChange) {
+        onChange(null);
+      }
+    },
+    [onChange],
+  );
+
+  const deleteButton = (
```
Comment (nicolabosco87) [inline src/components/layout/SideBar/Overview/OverviewFilters/AssigneeFilter/AssigneeFilter.tsx:45]: Maybe export `deleteButton` code into a dedicated shared component?

File: src/components/shared/ButtonGroupMultiselect/ButtonGroupMultiselect.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+          onChange(newState, event);
+        }
+      },
+    [onChange, options, values],
+  );
+
+  const isSelected = (value: T) => values.indexOf(value) >= 0;
+
+  return (
+    <Group className={classes.root} gap={0} wrap="nowrap">
```
Comment (nicolabosco87) [inline src/components/shared/ButtonGroupMultiselect/ButtonGroupMultiselect.tsx:78]: Button.Group component could be used here

File: src/components/layout/SideBar/Overview/Overview.tsx [hunk 3]
```diff
@@ -25,8 +72,148 @@ export const Overview: React.FC<{
 
   const openSidebar = () => toggle(SIDEBAR_STATUS.OPEN);
 
+  const expandCollapseClick = useCallback(() => {
+    setExpanded((expanded) => !expanded);
+  }, [setExpanded]);
+
+  const filterCases = (cases: TCase[], filters: TOverviewFilters) => {
```
[the remark this answers is not in the document]
Comment (kino90) [reply src/components/layout/SideBar/Overview/Overview.tsx]: Took this one step ahead, moved all filtering logics to a `useOverviewFilters` hook and moved the filter function to a `overviewFilters.utils.ts` file

File: src/components/layout/SideBar/Overview/OverviewFilters/AssigneeFilter/AssigneeFilter.tsx (hunk 1 above)
Comment (kino90) [reply src/components/layout/SideBar/Overview/OverviewFilters/AssigneeFilter/AssigneeFilter.tsx]: done

File: src/components/shared/ButtonGroupMultiselect/ButtonGroupMultiselect.tsx (hunk 2 above)
Comment (kino90) [reply src/components/shared/ButtonGroupMultiselect/ButtonGroupMultiselect.tsx]: done :)

**Comment by kino90** [comment]
Ooof, conflicts! I'll solve em after lunch

**Review by nicolabosco87 — APPROVED** [review]

**Pull request merged**